frontend/src/components/AssignmentModal/ContactInfoCard.vue

```diff
@@ -0,0 +1,418 @@
+            :is-required="true"
+            :label="'Zip code*'"
+            :placeholder="'XXXXX-XXXX'"
+          />
+        </v-col>
+      </v-row>
+      <v-row class="pb-7">
+        <v-col cols="12">
+          <v-btn
+            color="info"
+            variant="elevated"
+            @click="() => addInfo(contactData.id, 'address')"
+          >
+            + Add address
+          </v-btn>
+        </v-col>
+      </v-row>
+      <v-divider 
+        thickness="5"
+        color="primary"
+      />
+    </v-container>
+
+    <v-container>
+      <v-row>
+        <v-col cols="12">
+          <v-btn
+            color="primary"
+            variant="elevated"
+            @click="addContact"
+          >
+            + Add new contact
+          </v-btn>
+        </v-col>
+      </v-row>
+    </v-container>
+  </v-card-text>
+</template>
+
+<script setup lang="ts">
+  import { defineEmits, reactive, computed, onMounted } from 'vue';
+  import { storeToRefs } from 'pinia';
+
+  import { type IFormRules } from '@/utils/interfaces';
+  import { useAssignmentStore } from '@/stores/assignment';
+  import { useConfirmationStore } from '@/stores/confirmation';
+
+  import CustomInput from '@/components/UI/CustomInput.vue';
+  import CustomSelect from '@/components/UI/CustomSelect.vue';
+  import CustomAutocomplete from '@/components/UI/CustomAutocomplete.vue';
+
+  import { 
+    contactTypesLabels,
+    phoneNumberTypesLabels, 
+    addressesTypesLabels, 
+    statesCodesLabels 
+  } from '@/helpers/assignmentModal';
+
+  const { assignmentState } = storeToRefs(useAssignmentStore());
+  const { 
+    addContactData, 
+    addContactInfoData,
+    removeContactData,
+    removePhoneNumberData,
+    removeAddressData
+  } = useAssignmentStore();
+  const { setConfirmationDataAndShow } = useConfirmationStore();
+
+  const emits = defineEmits(['validate-form']);
+
+  onMounted(() => {
+    emits('validate-form');
+  })
+
+  const getContactTypesLabels = computed(
+    () => contactTypesLabels.filter(contactType => !assignmentState.value.contactsInfo.some(
+        contact => contact.type?.includes(contactType) && 
+        (contact.type.toLowerCase().includes('owner') ||
+        contact.type.toLowerCase().includes('driver') ||
+        contact.type.toLowerCase().includes('estimator'))
+      ))
+  );
+
+  const setAssignmentDataPropName = (infoType: string) => {
+    const infoTypeInLowerCase = infoType.toLowerCase();
+
+    if (infoTypeInLowerCase.includes('phone')) {
+      return 'phoneNumbers';
+    } else if (infoTypeInLowerCase.includes('address')) {
+      return 'addresses';
+    }
+
+    return '';
+  }
+
+  const addContact = () => {
+    const sortedContactsArr = 
+    assignmentState.value.contactsInfo.sort((a,b) => a.id - b.id);
+
+    const qtyOfContacts = sortedContactsArr.length;
+    const lastContactIdValue = sortedContactsArr[qtyOfContacts - 1].id;
+
+    addContactData({
+      id: lastContactIdValue + 1,

```

Comment:
Try to avoid using incrementing of ids. If you need to create id use Web Crypto API https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API

Reply:
Done